\documentclass{beamer}

%% Use the UCLouvain latest theme
%% \usetheme{UCL2018}

%% Load Latex packages
\usepackage{amssymb}
\usepackage{float}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage[utf8]{inputenc}
\usepackage{lmodern}
\usepackage[authoryear, round]{natbib}
\usepackage{ragged2e}
\usepackage{wrapfig}
\usepackage{xmpincl}
\usepackage{xspace}

%% Colors
%% ------
\usepackage{color}
% Color panel used throughout the poster
\definecolor{lgray}{rgb}{0.9179688,0.9179688,0.9179688} % #ebebeb
\definecolor{dgray}{rgb}{0.796875,0.796875,0.796875} % #cccccc
\definecolor{vdgray}{rgb}{0.3984375,0.3984375,0.3984375} % #666666
\definecolor{coral}{rgb}{0.9960938,0.4960938,0.3125000} % #ff7f50
\definecolor{blue}{rgb}{0.4218750,0.6484375,0.8007812} % #6ca6cd
\definecolor{green}{rgb}{0.6992188,0.7265625,0.5078125} % #b3ba82
\definecolor{yellow}{rgb}{0.9570312,0.8671875,0.6992188} % #f5deb3


%% Coding fonts
%% ------------
%% Font for R chunks 
\usepackage{listings} 
\lstset{
  language=R,
  basicstyle=\footnotesize\ttfamily\color{vdgray}, % the size of the fonts that are used for the code
  numbers=left,                   % where to put the line-numbers
  numberstyle=\tiny\color{gray},  % the style that is used for the line-numbers
  stepnumber=1,                   % the step between two line-numbers.
  numbersep=0.1cm,                % how far the line-numbers are from the code
  backgroundcolor=\color{lgray},  % choose the background color. You must add \usepackage{color}
  deletekeywords={stat, model, matrix},
  keywordstyle=\color{blue},      % keyword style
  stringstyle=\color{green},      % string literal style
  xleftmargin=0.5cm,
}
%% Create command for highlighting inline code or variables
\newcommand{\hcode}[2][lgray]{{\ttfamily\color{vdgray}\colorbox{#1}{#2}}}

\newcommand{\frametitlesection}[1]{\frametitle{\centering #1 \footnotesize \hspace{0pt plus 1 filll} \insertsection}}

%% Presentation metadata
\title{Replicate Specht et al. 2019 mass spectrometry-based single-cell
proteomics analysis}
\author[]{Laurent Gatto, Christophe Vanderaa}
\date{18 August 2020}
\institute[]{CBIO, de Duve Institute, UCLouvain}
%%\subtitle{Slides available at: \url{}}

\begin{document}
    
\pdfinfo {/Author(Laurent Gatto - UCLouvain)}
    
%-----------------------------------------------
% Abstract
%-----------------------------------------------
        
%% Abstract: Recent advances in sample preparation, processing and mass
%% spectrometry (MS) have allowed the emergence of MS-based single-cell
%% proteomics (SCP). We present a robust and standardized workflow to reproduce
%% the data analysis of SCoPE2, one of the pioneering works in the field
%% developed by the Slavov Lab. The implementation uses well-defined 
%% Bioconductor classes that provide powerful tools for single-cell RNA
%% sequencing and for MS-based proteomics. We demonstrate that our pipeline can
%% reproduce the SCoPE2 analysis using only a few lines of code.

%-----------------------------------------------
% Title page
%-----------------------------------------------

\begin{frame}[plain]
\titlepage
\end{frame}

%-----------------------------------------------
% Table of content
%-----------------------------------------------


\AtBeginSection[]{
  \setbeamercolor{background canvas}{bg=cyan}
  \setbeamercolor{section in toc}{fg=black}
  \setbeamerfont{section in toc}{size=\large}

  \mode<handout>{
    \setbeamercolor{background canvas}{bg=white}
    \setbeamercolor{section in toc}{fg=black}
  }

  \begin{frame}[plain]
    \frametitle{Outline}
    \tableofcontents[currentsection,hideothersubsections]
  \end{frame}

  \setbeamercolor{background canvas}{bg=white}
}


%-----------------------------------------------
% Introduction
%-----------------------------------------------

\section{Introduction}

\begin{frame}
    \frametitlesection{State of the art in MS-SCP}

    MS-SCP: Mass spectrometry-based single-cell proteomics 
    
    MS-SCP consist of shotgun proteomics at single-cell level 
    
    \begin{itemize}
        \item{SCoPE2 quantifies thousands of proteins x thousands single-cells}
        \item{Full protocole available}
        \item{Full analysis script available}
    \end{itemize}
    
    \textbf{BUT}
    
    Lack of standardized analysis software
\end{frame}

\begin{frame}
    \frametitlesection{Our objectives}

    Provide a suite of software package dedicated to MS-SCP that fulfill:

    \begin{itemize}
        \item{User-friendly}
        \item{Computationaly efficient}
        \item{Modularity: integrate other software packages}
        \item{Promote reproducibility}
        \item{Platform-independent}
        \item{Free of charge}
    \end{itemize}
    
    R/Bioconductor is an ideal environment
    
\end{frame}


%-----------------------------------------------
% Data framework
%-----------------------------------------------

\section{Data framework}

\begin{frame}
    \frametitlesection{SingleCellExperiment}
    
    
    \begin{columns}
        \begin{column}{0.8\textwidth}
            SingleCellExperiment: provides dedicated framework for single-cell 
            data analysis. 
            
            Available on Bioconductor.
        \end{column}
        \begin{column}{0.2\textwidth}
            \includegraphics[width=\linewidth]{figs/sticker_SingleCellExperiment.png}
        \end{column}
    \end{columns}
    
    \includegraphics[width=0.9\linewidth]{figs/SingleCellExperiment.png}
    
\end{frame}

\begin{frame}
    \frametitlesection{QFeatures}
    
    \centering
    
    \begin{columns}
        \begin{column}{0.8\textwidth}
            QFeatures: data framework dedicated to manipulate and process 
            MS-based quantitative data. 
            
            Submitted to Bioconductor.
        \end{column}
        \begin{column}{0.2\textwidth}
            \includegraphics[width=\linewidth]{figs/sticker_QFeatures.png}
        \end{column}
        
    \end{columns}
    
    \includegraphics[width=0.8\linewidth]{figs/QFeatures.png}
    
\end{frame}


\begin{frame}
    \frametitlesection{Data framework}

    MS-SCP data framework = SingleCellExperiment + QFeatures
    \begin{figure}
        \centering
        \includegraphics[width=.9\linewidth]{figs/SCP_framework.png}
    \end{figure}

    
\end{frame}

\begin{frame}
    \frametitlesection{MS-SCP package suite}
    
    scpdata: distributes published MS-SCP datasets (e.g. SCoPE2 dataset)
    
    scp: provides functionality for manipulating the MS-SCP data structure

\end{frame}

%-----------------------------------------------
% Standardized workflow
%-----------------------------------------------

\section{Standardized workflow}


\begin{frame}[fragile]
    \frametitlesection{Load data}
    
    Load the SCoPE2 dataset called \hcode{specht2019v2}
    
    \begin{lstlisting}
library(scpdata)
data("specht2019v2")
    \end{lstlisting}
    
    Dataset overview
    
    \begin{lstlisting}
show(specht2019v2)
    \end{lstlisting}
    
    \begin{lstlisting}[language = TeX, numbers = none, basicstyle = \tiny\ttfamily\color{vdgray}]
An instance of class QFeatures containing 179 assays:
 [1] 190222S_LCA9_X_FP94AA: SingleCellExperiment with 2823 rows and 11 col...
 [2] 190222S_LCA9_X_FP94AB: SingleCellExperiment with 4297 rows and 11 col...
 [3] 190222S_LCA9_X_FP94AC: SingleCellExperiment with 4956 rows and 11 col...
 ...
 [177] 191110S_LCB7_X_APNOV16plex2_Set_9: SingleCellExperiment with 4626 r...
 [178] peptides: SingleCellExperiment with 9208 rows and 1018 columns
 [179] proteins: SingleCellExperiment with 2772 rows and 1018 columns
    \end{lstlisting}
\end{frame}


\begin{frame}[fragile]
    \frametitlesection{Feature selection}
    
    Filter out features based on the feature metadata
    
    \bigskip
    
    Example: filter out reverse hits. The filter is applied to the 
    \hcode{Reverse} field in the feature metadata

    \begin{lstlisting}
filterFeatures(specht2019v2, 
               ~ Reverse != "+")
    \end{lstlisting}
    
    Source code in \hcode{QFeatures}
\end{frame}

\begin{frame}[fragile]
    \frametitlesection{QC metrics (1)}
    \small
    Interesting metrics for MS-SCP quality control: 
    
    \begin{itemize}
        \item{Sample to carrier ratio}: ratio of the carrier channel intensity 
        signal over the sample channel intensity
        \item{Peptide FDR\footnote{false discovery rate}: expected rate of 
        wrongly assigned features to a given peptide}
        \item{Cell median CV\footnote{coefficient of variation}: reliability of 
        the protein quantification summarized over each cell.}
    \end{itemize}
    
    Example:

    \begin{lstlisting}
computeMedianCV(specht2019v2,
                i = "peptides", 
                proteinCol = "protein", 
                peptideCol = "peptide", 
                batchCol = "Set")
    \end{lstlisting}
    
    Source code in \hcode{scp}
    
\end{frame}


\begin{frame}[fragile]
    \frametitlesection{QC metrics (2)}

    QC metrics are stored in the data set for plotting or subsetting
    
    \begin{lstlisting}[basicstyle = \scriptsize\ttfamily\color{vdgray}]
library(tidyverse)
specht2019v2[["peptides"]] %>%
  colData %>%
  data.frame %>%
  ggplot(aes(x = MedianCV, 
             fill = SampleType)) +
  geom_histogram() +
  geom_vline(xintercept = 0.4)
    \end{lstlisting}
    
    \begin{figure}
        \centering
        \includegraphics[width=.7\linewidth]{figs/medianCV.png}
    \end{figure}
\end{frame}

\begin{frame}[fragile]
    \frametitlesection{Feature aggregation}
    Feature aggregation includes 2 steps:
    \begin{itemize}
        \item{Combine the quantiative data from multiple features to a single 
        aggregated features}
        \item{Store the relationship between the parent features and the 
        aggregated features}
    \end{itemize}
    
    Example: aggregate peptides to proteins
    
    \begin{lstlisting}
aggregateFeatures(specht2019v2,
                  i = "peptides",
                  name = "proteins", 
                  fcol = "protein", 
                  fun = colMedians, na.rm = TRUE)
    \end{lstlisting}
    
    Source code in \hcode{QFeatures}
    
\end{frame}


\begin{frame}[fragile]
    \frametitlesection{Managing missingness}
    
    \hcode{0}'s can be either \textbf{biological} or \textbf{technical} zero. 
    They are better relaced by \hcode{NA}'s.
    
    \begin{lstlisting}
zeroIsNA(specht2019v2,
         i = "peptides")
    \end{lstlisting}
    
    Features containing too many missing data (e.g. >= 99 \%) should be removed
    
    \begin{lstlisting}
filterNA(specht2019v2, 
         i = "peptides", 
         pNA = 0.99)
    \end{lstlisting}
    
    Source code in \hcode{QFeatures}
    
\end{frame}

\begin{frame}[fragile]
    \frametitlesection{Data transformation}
    
    Common data transformation can easily be applied:
    
    \begin{itemize}
        \item{Normalization}
        \item{Log-transformation}
        \item{Imputation}
    \end{itemize}
    
    Example: $log_2$-transformation:
    
    \begin{lstlisting}
logTransform(specht2019v2, 
             i = "peptides", 
             base = 2,
             name = "peptides_log")
    \end{lstlisting}
    
    Source code in \hcode{QFeatures}
    
\end{frame}

\begin{frame}[fragile]
    \frametitlesection{Custom functions}
    
    Some custom function can be applied to the data set too. 
    
    Example: batch correction using \hcode{sva::ComBat}. First, extract the data 
    to correct
    
    \begin{lstlisting}[basicstyle = \scriptsize\ttfamily\color{vdgray}]
sce <- specht2019v2[["proteins"]]
    \end{lstlisting}
    
    Build the correction matrix and apply the ComBat algorithm
    
    \begin{lstlisting}[basicstyle = \scriptsize\ttfamily\color{vdgray}]
batch <- colData(sce)$Set
model <- model.matrix(~ SampleType, data = colData(sce))
assay(sce) <- ComBat(dat = assay(sce), 
                     batch = batch, 
                     mod = model)
    \end{lstlisting}
    
    Add the corrected protein to the dataset and keep feature relationships
    
    \begin{lstlisting}[basicstyle = \scriptsize\ttfamily\color{vdgray}]
addAssay(specht2019v2,
         sce,
         name = "proteins_batchC") %>%
addAssayLinkOneToOne(from = "proteins",
                     to = "proteins_batchC")
    \end{lstlisting}
    
    
\end{frame}

%-----------------------------------------------
% Replicating SCoPE2
%-----------------------------------------------

\section{Replicating SCoPE2}

\begin{frame}
    \frametitlesection{Benchmarking data matrices}
    \begin{columns}
        \begin{column}{0.5\textwidth}
            \textbf{Peptides} \\
            \includegraphics[width=\linewidth]{figs/Benchmark_pep_venn.png} \\
            \includegraphics[width=\linewidth]{figs/Benchmark_pep_err.png}
        \end{column}
        \begin{column}{0.5\textwidth}
            \textbf{Proteins} \\
            \includegraphics[width=\linewidth]{figs/Benchmark_prot_venn.png} \\
            \includegraphics[width=\linewidth]{figs/Benchmark_prot_err.png}
        \end{column}
    \end{columns}
    
\end{frame}

\begin{frame}
    \frametitlesection{Replicate figures from SCoPE2 (1)}
    
    
\end{frame}

\begin{frame}
    \frametitlesection{Replicate figures from SCoPE2 (2)}
    
\end{frame}

\begin{frame}
    \frametitlesection{Replicate figures from SCoPE2 (3)}
    
\end{frame}

%-----------------------------------------------
% Conclusion 
%-----------------------------------------------

\section{Conclusion}

\begin{frame}
    \frametitlesection{Take home message}
    
    \begin{itemize}
        \item{\hcode{scp} package suite provides a standardized environment for 
        performing MS-SCP data analysis}
        \item{Flexibly reproduce existing analyses from different groups or 
        protocoles (multiplex vs label free)}
    \end{itemize}
    
    \bigskip
    
    Advantages:
    
    \begin{itemize}
        \item{Allow automation of the analysis}
        \item{Facilitate new computational developments}
        \item{Promotes reproducibility}
        \item{Increases field visibility}
        \item{Include other modalities: scRNA-Seq, ATAC-Seq, etc}
    \end{itemize}
    
\end{frame}

\begin{frame}
    \frametitlesection{Resources}
    
    \textbf{Packages}
    
    \begin{itemize}
        \item{\hcode{scp}: GitHub repository \hcode{UClouvain-CBIO/scp}}
        \item{\hcode{scpdata}: coming soon
        \item{\hcode{QFeatures}: GitHub repository \hcode{rformassspectrometry/QFeatures}}
        \item{\hcode{SingleCellExperiment}: Bioconductor}
    \end{itemize}
    
    \bigskip
    
    \textbf{SCoPE2 reproduction vignette}
    
    Available at...
    
    \bigskip
    
    \textbf{Slides and source code}
    
    Available at...
    
\end{frame}

\begin{frame}
    \frametitlesection{Acknowledgements}
    
\end{frame}

\end{document}